### ISSUE TITLE
Add price to TradeOrder

### ISSUE THREAD

**Issue by Janaka-Steph** [tdex-network/tdex-sdk#134]
Currently we call `discoverer.discover({ asset, amount: sats })` to get the best order, then we have to call MarketPrice again to get the price preview. Instead `TradeOrder` could include the price.

@tiero @louisinger

**Comment by louisinger** [comment]
nACK: price is not a "constant from the Daemon point of view". It means that a price at time t can be different than the price at time t+1. That's why we must call the preview price just before making the swap (in order to be sure we'll trade at the right price)

**Comment by tiero** [comment]
Since we are sending the payload `asset` and `amount` already and with `bestPrice` strategy we do NOT only need to call Balances endpoint, but also some info from client, we basically doing already the preview when picking a provider&market combo: I do think it's fine to return the price right away, if the user is making a trade just right after. If it fails for bad pricing, he then call again as normal

**Comment by Janaka-Steph** [comment]
Afaik with bestPrice we already compute price for each order

**Comment by louisinger** [comment]
> Afaik with bestPrice we already compute price for each order

Right. But `Discoverer` is an abstraction of discovery functions. it must work with any Discovery, including the ones without `marketPrice` requests (like `bestBalance` only for instance). I really don't know how to handle this properly in `Discoverer` abstraction. 

I think in case of **only bestPrice** is used, better to do not use `Discoverer` and write your own "abstraction" caching the `marketPrice` requests
